### DISCUSSION TITLE
Problem with message_queue redis database number

### DISCUSSION THREAD

**Discussion by hem4nt** [miguelgrinberg/Flask-SocketIO#1615] [Q&A]
I have two flask_socketio apps running which use `message_queue="redis://localhost:6379/2"`  and `message_queue="redis://localhost:6379/1"` respectively. 
But it looks like I'm doing it wrong or it is not honouring the database numbers `2` and `1` but may be using `0` as default because when I `broadcast` some message from an external process, it gets broadcast from both the apps!

In the external processes I initialise the app like this:
```
   socketio = SocketIO()
   socketio.init_app(None, message_queue=f"redis://localhost:6379/{db_number}")
```
Can you please help?

**Comment by miguelgrinberg** [comment]
I'm not sure, because the URL that you provide is passed intact to the redis package. Any chance your `db_number` variable is zero or empty?

**Comment by hem4nt** [comment]
When I create the redis object I do it as below and it takes db number as a separate argument. So passing the URL intact is not enough? I remember it ignoring the db number in the URL when I had tried it earlier.
```
from redis import Redis
from app.app_config import config


def create_redis():
    redis = Redis(config['REDIS']['REDIS_HOSTNAME'],
                  db=config['REDIS']['REDIS_DB_NUM'])
    return redis
```

**Comment by miguelgrinberg** [comment]
@hem4nt I don't understand what is this code you are showing me. This package creates its own Redis connections from the URL that you pass, so what is the significance of your own method of creating Redis connections?

**Comment by hem4nt** [comment]
@miguelgrinberg, Sorry that code is from my app but not related to this package. I brought that here to point out that `Redis()` takes an argument `db` for db number. So I guess flask_socketio package must be extracting the db number from the `message_queue` arguement and pass separately as `db` to `Redis()` and passing the url alone intact may not cause Redis() to point to the correct db number. 
I'm sure I'm passing correct url to `socketio.init_app()` and couldn't think of any other problem.

**Comment by miguelgrinberg** [comment]
No, Flask-SocketIO passes the URL to Redis. It is Redis that parses it to obtain the database number. See the [Redis.from_url](https://redis-py.readthedocs.io/en/stable/#redis.ConnectionPool.from_url) function.
